go/internal/controller/predicates/discovery_disabled_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+func TestDiscoveryDisabledPredicate(t *testing.T) {
+	predicate := DiscoveryDisabledPredicate{}
+
+	tests := []struct {
+		name     string
+		labels   map[string]string
+		expected bool
+	}{
+		{
+			name:     "no label - should process",
+			labels:   nil,
+			expected: true,
+		},
+		{
+			name:     "discovery label set to disabled - should not process",
```

Comment:
What happens in the case of a resource that has already been discovered? Will this prevent it from being "undiscovered"?